framework/src/org/checkerframework/common/util/count/TypeStatistics.java

```diff
@@ -0,0 +1,79 @@
+package org.checkerframework.common.util.count;
+
+import com.sun.source.tree.ArrayAccessTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.NewArrayTree;
+import com.sun.source.tree.ParameterizedTypeTree;
+import javax.annotation.processing.SupportedSourceVersion;
+import javax.lang.model.SourceVersion;
+import org.checkerframework.framework.source.SourceChecker;
+import org.checkerframework.framework.source.SourceVisitor;
+import org.checkerframework.javacutil.AnnotationProvider;
+
+/**
+ * An annotation processor for listing the facts about types and array accesses locations of
+ * annotations. To invoke it, use
+ *
+ * <pre>
+ * javac -proc:only -processor org.checkerframework.common.util.count.TypeStatistics <em>MyFile.java ...</em>
+ * </pre>
+ */
+@SupportedSourceVersion(SourceVersion.RELEASE_8)
+public class TypeStatistics extends SourceChecker {
+
+    int generics = 0;
```

Comment:
Helpful to document what is being counted.  (Less critical, but still useful here, once there is class javadoc.)